compiler/rustc_target/src/spec/armv7_unknown_linux_uclibceabihf.rs

```diff
@@ -0,0 +1,23 @@
+use crate::spec::{Target, TargetOptions};
+
+// This target is for uclibc Linux on ARMv7 without NEON or
+// thumb-mode. See the thumbv7neon variant for enabling both.
+
+pub fn target() -> Target {
+    let base = super::linux_uclibc_base::opts();
+    Target {
+        llvm_target: "armv7-unknown-linux-gnueabihf".to_string(),
```

Comment:
I actually don't know what the implications of the LLVM target triple are here.  The code in `armv5te_unknown_linux_uclibceabi.rs` uses the same `gnueabihf` suffix, for what that's worth.  I made this requested change, but compilation now fails with:
```          
/home/vagrant/armv7-eabihf--uclibc--bleeding-edge-2020.08-1/bin/../lib/gcc/arm-buildroot-linux-uclibcgnueabihf/10.2.0/../../../../arm-buildroot-linux-uclibcgnueabihf/bin/ld: failed to merge target specific data of file /home/vagrant/rust/build/x86_64-unknown-linux-gnu/stage1-std/armv7-unknown-linux-uclibceabihf/release/deps/liballoc-b07e17534c57ab21.rlib(alloc-b07e17534c57ab21.alloc.674e4019-cgu.14.rcgu.o)
```

I note that uclibceabihf isn't a known `EnvironmentType` listed here: https://llvm.org/doxygen/classllvm_1_1Triple.html

Reply:
I see, gnueabihf is a okay fallback to use in that case. Otherwise LLVM will use an `unknown` and that's why you'll see mismatches like these come up.